### PR TITLE
docs: Add API reference

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds API reference documentation.

### How has this code been tested

N/A

### Special notes for your reviewer

N/A

### CODE REVIEW

File: docs/api-references.md [hunk 1]
```diff
@@ -0,0 +1,1373 @@
...
+- [fleet.azure.com/v1alpha1](#fleetazurecomv1alpha1)
+- [placement.kubernetes-fleet.io/v1beta1](#placementkubernetes-fleetiov1beta1)
+
+
+## cluster.kubernetes-fleet.io/v1beta1
+
+
+
+### Resource Types
+- [InternalMemberCluster](#internalmembercluster)
```
Comment (ryanzhang-oss) [inline docs/api-references.md:14]: please only include external facing API

Comment (michaelawyu) [reply docs/api-references.md:14]: Sure👌

Comment (michaelawyu) [reply docs/api-references.md:14]: I will hide `InternalMemberCluster` and `AppliedWorks`.

Comment (michaelawyu) [reply docs/api-references.md]: Also removed reference to v1alpha1 APIs.

**Comment by michaelawyu** [comment]
Hi @britaniar ! Thanks so much for the comments. The tricky part is, this is an auto-generated file; the contents are pulled from the comments in the source code through the journey of development, and to fix them we would need a separate PR to update the code, the CRDs, and release them first, which is a little bit difficult to follow through considering the timeframe and CCOA restrictions. If possible, I could keep your comments in a GitHub issue, and make the required edits when we are out of the embargo; would this be OK?

**Comment by britaniar** [comment]
> Hi @britaniar ! Thanks so much for the comments. The tricky part is, this is an auto-generated file; the contents are pulled from the comments in the source code through the journey of development, and to fix them we would need a separate PR to update the code, the CRDs, and release them first, which is a little bit difficult to follow through considering the timeframe and CCOA restrictions. If possible, I could keep your comments in a GitHub issue, and make the required edits when we are out of the embargo; would this be OK?

I think that would be okay!

File: docs/api-references.md [hunk 2]
```diff
@@ -0,0 +1,829 @@
...
+- [placement.kubernetes-fleet.io/v1beta1](#placementkubernetes-fleetiov1beta1)
+
+
+## cluster.kubernetes-fleet.io/v1beta1
+
+
+
+### Resource Types
+- [MemberCluster](#membercluster)
+- [MemberClusterList](#memberclusterlist)
```
Comment (ryanzhang-oss) [inline docs/api-references.md:14]: do we need this?

File: docs/api-references.md [hunk 3]
```diff
@@ -0,0 +1,829 @@
...
+
+
+
+### Resource Types
+- [ClusterResourceBinding](#clusterresourcebinding)
+- [ClusterResourcePlacement](#clusterresourceplacement)
+- [ClusterResourceSnapshot](#clusterresourcesnapshot)
+- [ClusterSchedulingPolicySnapshot](#clusterschedulingpolicysnapshot)
+- [Work](#work)
+- [WorkList](#worklist)
```
Comment (ryanzhang-oss) [inline docs/api-references.md:151]: Can we order this list by the importance of the object like put CRP first followed by Work and the rest more internal fields.

Comment (ryanzhang-oss) [inline docs/api-references.md:151]: can we move the list?

File: docs/api-references.md [hunk 4]
```diff
@@ -0,0 +1,829 @@
...
+
+
+
+
+#### ClusterResourcePlacement
+
+
+
+ClusterResourcePlacement is used to select cluster scoped resources, including built-in resources and custom resources, and placement them onto selected member clusters in a fleet. 
+ If a namespace is selected, ALL the resources under the namespace are placed to the target clusters. Note that you can't select the following resources: - reserved namespaces including: default, kube-* (reserved for Kubernetes system namespaces), fleet-* (reserved for fleet system namespaces). - reserved fleet resource types including: MemberCluster, InternalMemberCluster, ClusterResourcePlacement, ClusterSchedulingPolicySnapshot, ClusterResourceSnapshot, ClusterResourceBinding, etc.
```
[the remark this answers is not in the document]
Comment (ryanzhang-oss) [reply docs/api-references.md:231]: no, that's an internal object.

**Review by ryanzhang-oss — APPROVED** [review]
I need this link to link to Azure doc

**Pull request merged**